### PR TITLE
Server task6

### PR DESCRIPTION
**Pull request by dziq45**

### CODE REVIEW

File: server/service/taskValidator.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const taskModel = require('../model/taskModel');
+const { NotFoundError, UnauthorizedError } = require('../errorTypes/errorTypes')
+module.exports = {
+    validateUserTask: async (id, user) => {
```
Comment (time2funk) [inline server/service/taskValidator.js:4]: this can be stored as middleware

File: server/service/taskService.js [hunk 2]
```diff
@@ -21,25 +22,26 @@ module.exports = {
...
     deleteTask: async (id, user) => {
-        const filter = { _id: id, createdBy: user.name };
-        const result = await taskModel.deleteOne(filter);
-        return {
-            data: result
-        };
+        const filter = { _id: id };
+        await validateUserTask(id, user);
+        await taskModel.deleteOne(filter);
+        return
```
Comment (time2funk) [inline server/service/taskService.js:44]: guess what ?

File: server/service/taskService.js [hunk 3]
```diff
@@ -21,25 +22,26 @@ module.exports = {
...
         return {
             data: result
         };
 
     },
-    updateTask: async (id, params) => {
+    updateTask: async (id, params, user) => {
+        validatorService.validateAttributes(taskModel.attributes, params);
         const filter = { _id: id };
+        await validateUserTask(id, user)
```
Comment (time2funk) [inline server/service/taskService.js:34]: guess what ?

File: server/service/taskService.js [hunk 4]
```diff
@@ -10,8 +11,8 @@ module.exports = {
         };
     },
     saveTask: async (params) => {
+        validatorService.validateAttributes(taskModel.attributes, params)
```
Comment (time2funk) [inline server/service/taskService.js:14]: guess what ?

File: server/router/index.js [hunk 5]
```diff
@@ -3,25 +3,24 @@ const express = require('express');
...
-router.post('/api/task',authorizationJWT, taskController.addTask);
+router.post('/api/task', taskController.addTask);
 
-router.post('/api/task/:id',authorizationJWT,taskController.updateTask);
+router.post('/api/task/:id', taskController.updateTask);
 
-router.delete('/api/task/:id',authorizationJWT, taskController.deleteTask);
+router.delete('/api/task/:id', taskController.deleteTask);
 
-router.post('/api/register', userController.register);
```
Comment (time2funk) [inline server/router/index.js:21]: why ?

File: server/service/taskService.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 const taskModel = require('../model/taskModel');
 const Task = require('../db/models/Task');
-
+const validatorService = require('../service/validator')
```
Comment (time2funk) [inline server/service/taskService.js:3]: guess what ?

File: server/middlewares/index.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+const authorizationJWT = require('./auth');
+const handleErrors = require('./errorHandler');
+
+const endpointsConfig={
+    get:{
+        '/api/tasks':{
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/middlewares/index.js]: Done

File: server/middlewares/index.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    methods.forEach(method=>{
+        const paths = Object.keys(endpointsConfig[method]);
+        switch(method){
+            case 'get':
+                paths.forEach(path=>{
+                    const middlewares = endpointsConfig[method][path].middlewares;
+                    middlewares.forEach(middleware=>{
+                        app.get(path, middleware);
+                        console.log(middleware);
+                    })
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/middlewares/index.js]: Addded semicolon

File: server/middlewares/index.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+                })
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/middlewares/index.js]: Addded semicolon

File: server/service/taskValidator.js (hunk 1 above)
Comment (dziq45) [reply server/service/taskValidator.js]: Okay i store it in the middleware folder but it should be used like a normal function, not like a middleware, right?

File: server/service/taskService.js (hunk 2 above)
Comment (dziq45) [reply server/service/taskService.js]: Addded semicolon

File: server/service/taskService.js (hunk 3 above)
Comment (dziq45) [reply server/service/taskService.js:33]: Addded semicolon

File: server/service/taskService.js (hunk 4 above)
Comment (dziq45) [reply server/service/taskService.js]: Addded semicolon

File: server/service/taskService.js (hunk 6 above)
Comment (dziq45) [reply server/service/taskService.js:2]: Addded semicolon

File: server/router/index.js (hunk 5 above)
Comment (dziq45) [reply server/router/index.js:21]: These semicolons are hard ... but added this one back

File: server/service/userService.js [hunk 10]
```diff
@@ -26,7 +24,7 @@ module.exports = {
         };
     },
     signIn: async (credentials) => {
-        validatorService.validateAttributes(userModel.attributes, credentials);
+        //validatorService.validateAttributes(userModel.attributes, credentials);
         const user = await userModel.getByName(credentials.name)
```
Comment (time2funk) [inline server/service/userService.js:28]: semicolon please

File: server/service/taskService.js (hunk 6 above)
Comment (time2funk) [reply server/service/taskService.js:2]: are you sure

File: server/service/userService.js (hunk 10 above)
Comment (dziq45) [reply server/service/userService.js]: added

**Pull request merged**